README.md

```diff
@@ -83,6 +81,28 @@ class Weather < RubyLLM::Tool
 end
 
 chat.with_tool(Weather).ask "What's the weather in Berlin? (52.5200, 13.4050)"
+
+# Get structured output with JSON schema validation
+schema = {
+  type: "object",
+  properties: {
+    name: { type: "string" },
+    age: { type: "integer" },
+    interests: {
+      type: "array",
+      items: { type: "string" }
+    }
+  },
+  required: ["name", "age", "interests"]
+}
+
+# Returns a validated Hash instead of plain text
+user_data = chat.with_response_format(schema).ask("Create a profile for a Ruby developer")
+
+# Access the structured data using hash keys
+puts "Name: #{user_data.content['name']}"              # => "Jane Smith"
+puts "Age: #{user_data.content['age']}"                # => 32
+puts "Interests: #{user_data.content['interests'].join(', ')}"  # => "Ruby, Rails, API design"
```

Comment:
I feel like this is starting to get too verbose for the README. Let's leave it out of here. People can still find it in the docs.